src/main/java/com/felixgrund/codeshovel/changes/Yintroduced.java

```diff
@@ -22,4 +27,23 @@ public String toString() {
 				newFunction.getNameLineNumber()
 		);
 	}
+
+	@Override
+	public JsonObject toJsonObject() {
+		JsonObject obj = super.toJsonObject();
+		obj.addProperty("diff", getDiffAsString());
+		return obj;
+	}
+
+	private String getDiffAsString() {
+		if (this.diffString == null) {
+			try {
+				this.diffString = this.getDiffAsString("", newFunction.getSourceFragment());
+			} catch (IOException e) {
+				e.printStackTrace();
```

Comment:
Maybe you'll want to deal with this more gracefully? Is it ok if this happens sometimes? If it so happens, the output may be full of stack traces which can be really alarming. It might make sense to write a warning to the output. I'm not quite anymore sure if I used a proper logger in CodeShovel though...

Reply:
Oh I think I was trying to model it off of [this](https://github.com/ataraxie/codeshovel/blob/20a54f3237d2fc11a130d1442d282a964972ab9f/src/main/java/com/felixgrund/codeshovel/changes/Ycomparefunctionchange.java#L133). Do you want me to change this in both places?

Reply:
Oh yes please! I take the blame.